common/djangoapps/student/models.py

```diff
@@ -1602,6 +1602,11 @@ def enforce_single_login(sender, request, user, signal, **kwargs):    # pylint:
         else:
             key = None
         if user:
+            if not UserProfile.objects.filter(user=user):
```

Comment:
@justinabrahms here user name is used as profile name which is change able.
``user_profile = UserProfile(name=user.username, user=user)``

Done with adding ``.exists():``